Add 2025 report state of rust survey (#396)
rust-lang/surveys: surveys/2025/annual-survey/report/2026-02-13-2025-State-Of-Rust-Survey-results.md

| **Survey** | **Started** | **Completed** | **Completion rate** | **Views** |
|:----------:|------------:|--------------:|--------------------:|----------:|
|    2024    |       9 450 |         7 310 |               77.4% |    13 564 |
|    2025    |       9 389 |         7 156 |               76.2% |    20 397 |

Overall, the answers we received this year pretty closely match the results of last year, differences are often under a single percentage point. The number of respondents decreases slightly year over year. In 2025, we published multiple surveys (such as the [Compiler Performance][compiler-performance-survey] or [Variadic Generics survey][variadic-generics-survey]), which might have also contributed to less people answering this (longer) survey. We plan to discuss how (and whether) to combine the State of Rust survey with the ongoing work on the [Rust Vision Doc](https://blog.rust-lang.org/2025/12/03/lessons-learned-from-the-rust-vision-doc-process/).

Also to be noted that these numbers should be taken in context: we cannot extrapolate too much from a mere 7 000 answers and some optional questions have even less replies.

> Let's point out some interesting piece of data:

Reviewer: ```suggestion
Let's point out some interesting pieces of data:
```

- [Screenshotting Rust use](#screenshotting-rust-use)
- [Challenges and wishes about Rust](#challenges-and-wishes-about-rust)
- [Learning about Rust](#learning-about-rust)
- [Industry and community](#industry-and-community)

[variadic-generics-survey]: https://blog.rust-lang.org/inside-rust/2025/09/22/variadic-generics-micro-survey/
[compiler-performance-survey]: https://blog.rust-lang.org/2025/09/10/rust-compiler-performance-survey-2025-results/

## Screenshotting Rust use

> Confirmed that people develop using the stable compiler and keep up with releases, trusting our stability and compatibility guarantees. On the other hand, people use nightly out of "necessity" (for example, something not yet stabilized). Compared to last year ([link][report-2024]) we seem to have way many less nightly users. This may not be a significant data point because we are looking at a sliding window of releases and differences could depend on many factors (for example, at a specific point in time we might have more downloads of the nightly compiler because of a highly anticipated feature).

Reviewer: ```suggestion
Confirmed that people develop using the stable compiler and keep up with releases, trusting our stability and compatibility guarantees. On the other hand, people use nightly out of "necessity" (for example, something not yet stabilized). Compared to last year ([link][report-2024]) we seem to have way less nightly users. This may not be a significant data point because we are looking at a sliding window of releases and differences could depend on many factors (for example, at a specific point in time we might have more downloads of the nightly compiler because of a highly anticipated feature).

One example might be the very popular let chains and async closures features, which were stabilized last year.
```

### Why *not* using Rust?

We are interested and grateful to people *not* using Rust (anymore) when they tell us why they dropped the language. In most cases it seems to be a "see you again in the future" rather than a "goodbye".

<!-- chart: do-you-use-rust (height=400) -->
<!-- chart: why-did-you-stop-using-rust (height=400) -->

### Which unimplemented (or nightly only) features are you looking for to be stabilized?

We landed long-awaited features in 2025 (`let chains` and `async closures`) and are very often used. That's something to celebrate! Now in the charts `generic consts expressions` and `traits methods` are bubbling up are most-wanted new features. Most of the other nice-to-haves didn't change significantly.

<!-- chart: which-features-do-you-want-stabilized (height=400) -->

### Which of the following aspects of Rust present non-trivial problems to your programming productivity?

Little changes overall compared to 2024. The debugging story slipped from 2nd to 4th place (~2%). We will soon run a dedicated survey to learn more about it!

<!-- chart: which-problems-limit-your-productivity (height=400) -->

### How do you download crates to build Rust projects?

> TODO: This year we added a new entry (`git`). (Does it mean git checkout?)

Author: I don't know how to read this piece of data

> and if people actually find the output of [`--explain`] useful. Internal discussions hinted that we were not too sure about that but this graph contradicts our assumption. Good to know, we won't remove it :-)

Reviewer: ```suggestion
and if people actually find the output of [`--explain`] useful. Internal discussions hinted that we were not too sure about that but this graph contradicts our prior assumption. Seems like many Rust users actually do find compiler error code explanations useful.
```
I don't think we should be making such claims in the survey analysis blog posts.
